source/slang/slang-stdlib.cpp

```diff
@@ -203,7 +203,7 @@ namespace Slang
     static const OpInfo unaryOps[] = {
         { kIRPseudoOp_Pos,     "+",    ARITHMETIC_MASK },
         { kIROp_Neg,     "-",    ARITHMETIC_MASK },
-        { kIROp_Not,     "!",    ANY_MASK        },
+        { kIROp_Not,     "!",    BOOL_MASK | BOOL_RESULT },
```

Comment:
I don't think `BOOL_RESULT` is strictly needed if we have `BOOL_MASK`, since it means the operation would only be applied to types built up from `bool` anyway, right?